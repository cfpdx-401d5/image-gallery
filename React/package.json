{
  "name": "image-gallery",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "body-parser": "^1.17.1",
    "cors": "^2.8.1",
    "express": "^4.15.2",
    "mongoose": "^4.8.7",
    "morgan": "^1.8.1",
    "react": "^15.4.2",
    "react-dom": "^15.4.2"
  },
  "devDependencies": {
    "chai": "^3.5.0",
    "chai-http": "^3.0.0",
    "enzyme": "^2.7.1",
    "enzyme-to-json": "^1.5.0",
    "eslint": "^3.17.1",
    "eslint-plugin-react": "^6.10.0",
    "mocha": "^3.2.0",
    "nodemon": "^1.11.0",
    "react-addons-test-utils": "^15.4.2",
    "react-scripts": "0.9.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "start-server": "nodemon server.js",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  }
}
