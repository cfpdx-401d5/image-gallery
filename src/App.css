.App {
  text-align: center;
}


.App-header {
  background-color: #222;
  color: white;
}

div #root {
  text-align: center;
  width: 90%%;
  margin-left: auto;
  margin-right: auto;
  border: .5em solid #bc640b;
}

.viewer {
height: 80%;
border: .5em solid #000;
}

img {
  max-height: 100%;
  max-width: 100%;
  padding: 1em;
}