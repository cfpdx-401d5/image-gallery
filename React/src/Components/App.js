import React from 'react';

import CreateBunnyForm from './CreateBunnyForm';
import Selector from './Selector';
import View from './View';

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      viewStyle: null,
      babyBunBuns: [],
    };
    this.handleClick = this.handleClick.bind(this);
    this.deleteBunny = this.deleteBunny.bind(this);
    this.addBunny = this.addBunny.bind(this);
  }


  handleClick(viewValue) {
    console.log(viewValue);
    this.setState({
      viewStyle: viewValue
    });
  }

  deleteBunny(babyBunIndex, _id){
    fetch(`http://localhost:8000/bunniess/${_id}`, {method: 'DELETE'})
      .then(res => res.json())
      .then(res => {
          if(res.message) {
            //res.message;
            let babyBunBunsFiltered = this.state.babyBunBuns.filter((element, i)=>{
                if (i !== babyBunIndex) {
                    return element
                }
            });
            this.setState({
                babyBunBuns: babyBunBunsFiltered
            })
        }
    })
  }

  addBunny(newBunBun){
    fetch('http://localhost:8000/bunnies', {
      method: 'POST',
      mode: 'cors',
      headers: {
        'Accept': 'application/json',
        'Content-type': 'application/json'
      },
      body: JSON.stringify(newBunBun)
    })
      .then(res => res.json())
      .then(res => {
          delete res.__v;
          let babyBunBunsMultiplied = this.state.babyBunBuns.concat(res);
          this.setState({
              babyBunBuns: babyBunBunsMultiplied
          })
      })
  }

  render() {
    return (
      <div>
        <Selector 
          handleClick={this.handleClick}
        />
        <CreateBunnyForm 
          addBunny={this.addBunny}
        />
        <View 
          deleteBunny={this.deleteBunny} 
          babyBunBuns={this.state.babyBunBuns} 
          viewStyle={this.state.viewStyle}
        />
      </div>
    );
  }

  componentDidMount(){
    fetch('http://localhost:8000/bunnies')
      .then(res => res.json())
      .then(bunbuns => {
        this.setState(
          {babyBunBuns: bunbuns}
        )
      })
  }
}

