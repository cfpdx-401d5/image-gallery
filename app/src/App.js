import React, { Component } from 'react';

import ListDisplay from './displays/ListDisplay';
import ThumbnailDisplay from './displays/ThumbnailDisplay';
import GalleryDisplay from './displays/GalleryDisplay';
import DisplaySelector from './DisplaySelector';
import FormContainer from './image_form/FormContainer';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      displayType: 'list',
      images: [{
        id: 'id1',
        title: 'Cute Bunny',
        description: 'Isn\'t it fuzzy-wuzzy cutest thing you\'ve ever seen?',
        url: 'http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg'
      },
      {
        id: 'id2',
        title: 'Upside Down Sloth',
        description: 'Sloths just love to hang out.',
        url: 'http://kids.nationalgeographic.com/content/dam/kids/photos/animals/Mammals/Q-Z/sloth-beach-upside-down.jpg.adapt.945.1.jpg'
      },
      {
        id: 'id3',
        title: 'Waving Panda',
        description: 'When a panda waves, you wave back!',
        url: 'http://media4.s-nbcnews.com/j/newscms/2016_36/1685951/ss-160826-twip-05_8cf6d4cb83758449fd400c7c3d71aa1f.nbcnews-ux-2880-1000.jpg'
      }
      ],
    };
    this.setDisplay = this.setDisplay.bind(this);
    this.addImage = this.addImage.bind(this);
    this.deleteImage = this.deleteImage.bind(this);
  }

  setDisplay(whatDisplay) {
    this.setState({ displayType: whatDisplay });
  }

  addImage(e, imageSubmission) {
    e.preventDefault();
    // let newImgArray = [...this.state.images]; // copy the images[] in the state
    // newImgArray.push(imageSubmission); // make new image array

    // this.setState({ images: newImgArray });
    fetch('http://localhost:3001/images', {
      method: 'POST',
      mode: 'cors',
      headers: {
        'Accept': 'application/json',
        'Content-type': 'application/json'
      },
      body: JSON.stringify(imageSubmission)
    })
      .then(res => res.json())
      .then(res => {
        let newImgArray = [...this.state.images];
        delete res.__v;
        newImgArray.push(res);
        this.setState({ images: newImgArray });
      });
  }

  deleteImage(imageID) {
    // let newImgArray = this.state.images.filter(image => {
    //   return (image.id !== imageID);
    // });

    // this.setState({ images: newImgArray });
    fetch(`http://localhost:3001/images/${imageID}`, { method: 'DELETE' })
      .then(res => res.json())
      .then(res => {
        if (res.message) {
          let newImgArray = this.state.images.filter(image => {
            return (image.id !== imageID);
          });
          this.setState({ images: newImgArray });
        }
      });
  }

  render() {
    let DisplayType;

    if (this.state.displayType === 'list') DisplayType = ListDisplay;
    else if (this.state.displayType === 'thumbnail') DisplayType = ThumbnailDisplay;
    else if (this.state.displayType === 'gallery') DisplayType = GalleryDisplay;

    return (
        <div>
          <h1>Image Gallery</h1>
          <FormContainer addImage={this.addImage} />
          <DisplaySelector setDisplay={this.setDisplay} />
          <DisplayType images={this.state.images} deleteImage={this.deleteImage} />
        </div>
    );
  }

  componentDidMount() {
    fetch('http://localhost:3001/images')
      .then(res => res.json())
      .then(resImages => {
        this.setState({ images: resImages });
      });
  }
  
}

