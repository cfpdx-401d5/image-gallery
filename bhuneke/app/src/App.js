import React, { Component } from 'react';
import AddResource from './components/editing/AddResource';
import GalleryDisplay from './components/display/Gallery';
import ThumbnailDisplay from './components/display/Thumbnail';
import DetailDisplay from './components/display/Detail';
import fetcher from './helpers/fetcher';

const GALLERY_TYPE = 'gallery';
const LIST_TYPE = 'list';
const THUMBNAIL_TYPE = 'thumbnail';

export default class App extends Component {

    constructor(props) {
        super(props);
        this.state = {
            selectedView: '',
            resources: []
        };
        this.onViewSelect = this.onViewSelect.bind(this);
        this.onDelete = this.onDelete.bind(this);
        this.onAdd = this.onAdd.bind(this);
    }

    onViewSelect(e, selected) {
        e.preventDefault();
        this.setState({
            selectedView: selected
        });
    }
    doFetch() {
        fetcher({
            path: '/resources',
            method: 'GET',
        })
        .then(res => res.json())
        .then(resources => {
            this.setState({ resources });
        })
        .catch(err => { console.log(err); });
    }

    componentDidMount() {
        this.doFetch();
    }

    onDelete(id) {
        fetcher({
            path: `/resources/${id}`,
            method: 'DELETE'
        });
        this.doFetch();
    }

    onAdd(newResource) {
        let newData = this.state.resources.slice();
        newData.push(newResource);
        this.setState({
            resources: newData
        });
    }

    render() {
        return (
            <div>
                < ViewSelector onViewSelect={this.onViewSelect} />
                < ViewDisplay onDelete={this.onDelete} selectedView={this.state.selectedView} resources={this.state.resources} />
                < AddResource onAdd={this.onAdd} />
            </div>
        );
    }
}

function ViewSelector(props) {
    return (
        <div className='view-selector'>
            <button onClick={(e, selected) => props.onViewSelect(e, LIST_TYPE)}>List View</button>
            <button onClick={(e, selected) => props.onViewSelect(e, THUMBNAIL_TYPE)}>Thumbnail View</button>
            <button onClick={(e, selected) => props.onViewSelect(e, GALLERY_TYPE)}>Gallery View</button>
        </div>);
}

function ViewDisplay(props) {
    let selectedView = props.selectedView;
    if (selectedView) {
        return (
            <ListView
                selectedView={selectedView}
                resources={props.resources}
                onDelete={props.onDelete} 
            />);
    } else {
        return <p>Choose View!</p>;
    }
}

function ListView(props) {
    let DisplayType;
    const selectedView = props.selectedView;
    if (selectedView === LIST_TYPE) DisplayType = DetailDisplay;
    else if (selectedView === GALLERY_TYPE) DisplayType = GalleryDisplay;
    else if (selectedView === THUMBNAIL_TYPE) DisplayType = ThumbnailDisplay;

    const resourceList = props.resources.map(resource => {
        return (
            <DisplayType
                key={resource._id}
                resource={resource}
                onDelete={props.onDelete} 
            />);
    });

    return (
        <div>{resourceList}</div>
    );
}

export {
    ViewSelector,
    ViewDisplay,
    ListView,
};
