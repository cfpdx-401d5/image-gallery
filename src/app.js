import React from 'react';
import ImageView from './view/display-view';
import List from './view/detail-view';
import Gallery from './view/gallery-view';
import Thumbnail from './view/thumbnail-view';

import bunnies from './image';

const GALLERY_TYPE    = 'gallery';
const DETAIL_TYPE       = 'detail';
const THUMBNAIL_TYPE  = 'thumbnail';

function ViewSelector(props) {
    return (
      <div>
        <button onClick={() => props.onClick(GALLERY_TYPE)}>Gallery View</button>
        <button onClick={() => props.onClick(DETAIL_TYPE)}>List View</button>
        <button onClick={() => props.onClick(THUMBNAIL_TYPE)}>Thumbnail View</button>
      </div>
    );
}

function ImageView(props) {
    const bunnyArray = bunnies();
    const outputs = {
        [DETAIL_TYPE]: List,
        [GALLERY_TYPE]: Gallery,
        [THUMBNAIL_TYPE]: Thumbnail,
    };    
    
    const OutputComponent = outputs[props.view];

    return (
        <div>
            <ViewSelector onClick={props.onClick}/>
            {OutputComponent 
                ? <OutputComponent 
                    array={bunnyArray} />
                : <h3>Click a button to view bunnies!</h3>
            }          
        </div>
    );
}

export default class App extends React.Component {
    constructor(props) {
        super(props);
        this.handleClick = this.handleClick.bind(this);
        this.state = {
            view: null
        };
    }

    handleClick(value) {
        this.setState({
            view: value
        });
    }

    render() {
        return (
          <div>
            <ImageView view={this.state.view} onClick={this.handleClick}/>
          </div>
        );
    }
}
