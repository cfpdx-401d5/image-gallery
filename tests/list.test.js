/* eslint-disable */
import React from 'react';
import {shallow, render} from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../src/app';

describe('Full App', () => {
    it('full app display', () => {
        const rendered = shallow(<App />);
        expect(toJson(rendered)).toMatchSnapshot();
    });

    it.skip('list display', () => {
        const rendered = shallow(<App />);
        expect(toJson(rendered)).toMatchSnapshot();
    });

    it.skip('gallery display', () => {
        const rendered = shallow(<App />);
        expect(toJson(rendered)).toMatchSnapshot();
    });
    it.skip('thumbnail display', () => {
        const rendered = shallow(<App />);
        expect(toJson(rendered)).toMatchSnapshot();
    })
});
