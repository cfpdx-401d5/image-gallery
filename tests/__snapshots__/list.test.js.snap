exports[`list display list display 1`] = `<ul />`;
